components/windows_recall/windows_recall.cc

```diff
@@ -0,0 +1,64 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/windows_recall/windows_recall.h"
+
+#include <memory>
+
+#include "base/supports_user_data.h"
+#include "base/win/windows_version.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+
+namespace {
+
+struct WindowsRecallData : public base::SupportsUserData::Data {
+  static const int kKey = 0;  // Only address is used.
+
+  explicit WindowsRecallData(const bool enabled)
+      : is_windows_recall_enabled(enabled) {}
+
+  const bool is_windows_recall_enabled = false;
+};
+
+}  // namespace
+
+namespace windows_recall {
+
+bool IsWindowsRecallAvailable() {
+  return base::win::GetVersion() >= base::win::Version::WIN11;
```

Comment:
why do we have this method and also an enum with three states? That seems redundant?

Reply:
also this entire thing seems far more complicated than necessary. We never want this enabled so why not just override `SetInputScope` to always be in `IS_PRIVATE` 

in tsf_input_scope.cc
```
#define SetInputScope SetInputScope_ChromiumImpl

void SetInputScope(HWND window_handle, InputScope input_scope) {
  SetInputScope_ChromiumImpl(window_handle, IS_PRIVATE)
}
```

Reply:
I guess according to @ShivanKaul this does need to enabled in some cases